src/Compilers/Core/Portable/DiagnosticAnalyzer/SuppressMessageAttributeState.cs

```diff
@@ -284,7 +300,9 @@ private void DecodeGlobalSuppressMessageAttributes(Compilation compilation, ISym
         {
             Debug.Assert(symbol is IAssemblySymbol || symbol is IModuleSymbol);
 
-            var attributes = symbol.GetAttributes().Where(a => a.AttributeClass == this.SuppressMessageAttribute);
+            var attributes = symbol.GetAttributes().Where(a =>
+                a.AttributeClass == this.SuppressMessageAttribute
+                || a.AttributeClass == this.UnconditionalSuppressMessageAttribute);
             DecodeGlobalSuppressMessageAttributes(compilation, symbol, globalSuppressions, attributes);
```

Comment:
Could this be called with an argument with a null `AttributeClass`? I'm wondering if this API could give strange answers in the event that either SuppressMessageAttribute or UnconditionalSuppressMessageAttribute is null, and `a.AttributeClass` is null.

Reply:
Heh, yeah I suppose that's possible. I think that's an existing bug. I'd rather not hold up the PR for it if you don't mind, but I can file a bug about it.

Reply:
(I don't think it semantically matters because the diagnostic is only suppressed if the ID's match, but if the attribute doesn't exist, we won't be able to fetch the ID and they will never match).

Reply:
Filed https://github.com/dotnet/roslyn/issues/51972

Reply:
Sounds good.